tiledb/sm/c_api/tiledb.cc

```diff
@@ -5664,8 +5664,21 @@ int32_t tiledb_consolidation_plan_free_json_str(char** out) {
   return TILEDB_OK;
 }
 
-}  // namespace tiledb::api
+TILEDB_EXPORT int32_t tiledb_array_maximum_tile_size(
+    tiledb_array_t* array, uint64_t* max_in_memory_tile_size) {
+  api::ensure_array_is_valid(array);
+  api::ensure_output_pointer_is_valid(max_in_memory_tile_size);
+  try {
+    *max_in_memory_tile_size = array->array_->max_tile_size();
+  } catch (...) {
+    std::throw_with_nested(
+        api::CAPIStatusException("Failed to obtain max tile size "));
```

Comment:
This extra message doesn't seem to provide any new information. We already know it's failed, because we're processing an exception. And we already know we failed to load a maximum tile size. Seems better to leave catch and rethrow out.